### PR TITLE
Support next v15

### LINKED ISSUE #71: Next.js v15 support
Add official support for Next.js v15.

See [blog post](https://nextjs.org/blog/next-15) and [upgrade guide](https://nextjs.org/docs/app/building-your-application/upgrading/version-15).

### PR DESCRIPTION
**Pull request by stefee**
Fixes #71

See [blog post](https://nextjs.org/blog/next-15) and [upgrade guide](https://nextjs.org/docs/app/building-your-application/upgrading/version-15).

Changes:
- Bump `next` and `eslint-config-next` version to `15.0.0`
- Use async/await for `headers()`
- Add new examples for next v15 (copied from v14 examples, modified for async/await)

### CODE REVIEW

**Comment by amorey** [comment]
```
 ⚠ Invalid next.config.js options detected: 
 ⚠     Unrecognized key(s) in object: 'swcMinify'
 ⚠ See more info here: https://nextjs.org/docs/messages/invalid-next-config
 ```
 https://github.com/kubetail-org/edge-csrf/issues/71

**Comment by amorey** [comment]
in next15-approuter-html-submission:
```
We detected TypeScript in your project and reconfigured your tsconfig.json file for you.
   The following suggested values were added to your tsconfig.json. These values can be changed to fit your project's needs:

   	- target was set to ES2017 (For top-level `await`. Note: Next.js only polyfills for the esmodules target.)
 ```
 https://github.com/kubetail-org/edge-csrf/issues/71

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -2,3 +2,4 @@ packages:
   - "packages/**"
   - "shared"
   - "!examples/next14-approuter-html-submission/*"
+  - "!examples/next15-approuter-html-submission/*"
```
Comment (stefee) [inline pnpm-workspace.yaml:5]: Wasn't sure why this change was needed for the next14 example because I'm not particularly familiar with pnpm, but just duplicated the line to match for next15 anyway...

File: examples/next15-approuter-html-submission/next-env.d.ts [hunk 2]
```diff
@@ -2,4 +2,4 @@
 /// <reference types="next/image-types/global" />
 
 // NOTE: This file should not be edited
-// see https://nextjs.org/docs/basic-features/typescript for more information.
+// see https://nextjs.org/docs/app/building-your-application/configuring/typescript for more information.
```
Comment (stefee) [inline examples/next15-approuter-html-submission/next-env.d.ts:5]: The changes to `next-env.d.ts` are a result of running `pnpm dev` in each directory

**Comment by stefee** [comment]
`pnpm lint` and `pnpm build` seem to be failing for me on the v14 and v15 sentry examples @amorey FYI

**Comment by amorey** [comment]
what error are you getting when you run `pnpm lint` & `pnpm build` on the sentry examples?

**Comment by stefee** [comment]
It's possible I have something messed up with how I've installed the examples

<details>

<summary>View terminal output</summary>

```
> edge-csrf-example@0.1.0 lint /Users/Stef.Jones/src/edge-csrf/examples/next15-approuter-sentry
> next lint


./app/api/sentry-example-api/route.js
Error: Parsing error: [BABEL] /Users/Stef.Jones/src/edge-csrf/examples/next15-approuter-sentry/app/api/sentry-example-api/route.js: Requires Babel "^7.22.0", but was loaded with "7.18.0". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel. (While processing: "/Users/Stef.Jones/src/edge-csrf/node_modules/.pnpm/next@15.0.0_react-dom@18.2.0_react@18.2.0__react@18.2.0/node_modules/next/babel.js$3")

./app/global-error.jsx
Error: Parsing error: [BABEL] /Users/Stef.Jones/src/edge-csrf/examples/next15-approuter-sentry/app/global-error.jsx: Requires Babel "^7.22.0", but was loaded with "7.18.0". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel. (While processing: "/Users/Stef.Jones/src/edge-csrf/node_modules/.pnpm/next@15.0.0_react-dom@18.2.0_react@18.2.0__react@18.2.0/node_modules/next/babel.js$3")

./app/layout.tsx
1:37  Error: Missing semicolon.  @typescript-eslint/semi
2:41  Error: Missing semicolon.  @typescript-eslint/semi
3:23  Error: Missing semicolon.  @typescript-eslint/semi
5:44  Error: Missing semicolon.  @typescript-eslint/semi
10:2  Error: Missing semicolon.  @typescript-eslint/semi
21:4  Error: Missing semicolon.  @typescript-eslint/semi

./app/page.tsx
1:31  Error: Missing semicolon.  @typescript-eslint/semi
49:26  Error: Strings must use singlequote.  @typescript-eslint/quotes
55:25  Error: Strings must use singlequote.  @typescript-eslint/quotes
66:26  Error: Strings must use singlequote.  @typescript-eslint/quotes
72:25  Error: Strings must use singlequote.  @typescript-eslint/quotes
83:26  Error: Strings must use singlequote.  @typescript-eslint/quotes
89:25  Error: Strings must use singlequote.  @typescript-eslint/quotes
100:26  Error: Strings must use singlequote.  @typescript-eslint/quotes
106:25  Error: Strings must use singlequote.  @typescript-eslint/quotes
112:4  Error: Missing semicolon.  @typescript-eslint/semi

./app/sentry-example-page/page.jsx
Error: Parsing error: [BABEL] /Users/Stef.Jones/src/edge-csrf/examples/next15-approuter-sentry/app/sentry-example-page/page.jsx: Requires Babel "^7.22.0", but was loaded with "7.18.0". If you are sure you have a compatible version of @babel/core, it is likely that something in your build process is loading the wrong version. Inspect the stack trace of this error to look for the first entry that doesn't mention "@babel/core" or "babel-core" to see what is calling Babel. (While processing: "/Users/Stef.Jones/src/edge-csrf/node_modules/.pnpm/next@15.0.0_react-dom@18.2.0_react@18.2.0__react@18.2.0/node_modules/next/babel.js$3")

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules
 ELIFECYCLE  Command failed with exit code 1.
```

</details>

**Comment by amorey** [comment]
> It's possible I have something messed up with how I've installed the examples
> 
> View terminal output

The errors might be coming from running the linter in the same pnpm workspace as the package source. It passes for me when I run it outside the workspace so I think it's ok.

**Comment by amorey** [comment]
Can you merge my suggestions? https://github.com/stefee/edge-csrf/pull/1 Then I'll merge the PR into main.
